### PR TITLE
Proposal for encryption-at-rest (etcd data encryption)

### LINKED ISSUE #6393: Write proposal about encryption at rest for user clusters

### PR DESCRIPTION
**Pull request by embik**
Signed-off-by: Marvin Beckers <marvin@kubermatic.com>

**What this PR does / why we need it**:
This adds a proposal for encryption at rest, a feature in Kubernetes that allows encrypting data in etcd (primarily geared towards `Secrets`). encryption at rest is a security feature recommend by some Kubernetes security guidelines and users might need or want it for improving their security posture.

The core idea of the proposal is to provide the KMS integration for AWS, GCP and Azure as "turn-key" solutions if the cluster is running on those cloud providers, managing the KMS plugin configuration for users and supporting key rotation (automated key rotation is out of scope right now). A static encryption key scheme is supported too to cover clusters running on environments without KMS services.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #6393

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by embik** [comment]
Putting the hold label on this to make sure that there's enough room for discussing it before merging. Please feel free to comment and review anyway!

/hold

**Comment by kron4eg** [comment]
@embik since we are talking about security here, some threat modelling section would be very handy to have.

**Comment by embik** [comment]
@kron4eg thats a very good suggestion. I’m not sure I can draft a comprehensive threat modelling, but I’ll try to add something we can iterate on.

**Review by kron4eg — APPROVED** [review]

**Pull request merged**
